dynamic/go.mod

```diff
@@ -1,240 +0,0 @@
-module github.com/pulumi/pulumi-terraform-bridge/dynamic
-
-go 1.22.3
-
-replace (
-	// Use OpenTofu's fork of hcl (copied from OpenTofu).
-	github.com/hashicorp/hcl/v2 => github.com/opentofu/hcl/v2 v2.0.0-20240416130056-03228b26f391
```

Comment:
This rewrite is lost in the change. We can reintroduce it specifically scoped to the bin/pulumi-resource-terraform-provider build by using a Go workspace to build it, if necessary. Wanted to confirm if it is necessary.

Another option is to reintroduce it at top-level go.mod but it seems that would be an actual change to the bridge as currently `github.com/hashicorp/hcl/v2` is used elsewhere without the replace.